src/MessageRepository.php

```diff
@@ -26,4 +26,6 @@ public function retrieveAll(AggregateRootId $id): Generator;
      * @throws UnableToRetrieveMessages
      */
     public function retrieveAllAfterVersion(AggregateRootId $id, int $aggregateRootVersion): Generator;
+
+    public function paginate(int $perPage, PaginationCursor|null $cursor = null): Generator;
```

Comment:
There is no way to typehint that the generator needs to return an PaginationCursor, but it might be good to add it as a comment here.

Reply:
I've added a typehint 👍